src/main/java/com/coutemeier/maven/plugins/asciidoctor/lifecycle/util/WagonUtil.java

```diff
@@ -16,6 +43,154 @@
+            throw new MojoExecutionException( shortMessage );
+
+        } catch ( TransferFailedException e ) {
+            throw new MojoExecutionException( "Unable to configure Wagon: '" + repository.getProtocol() + "'", e );
+        }
+    }
+
+    /**
+     * Configure the Wagon with the information from serverConfigurationMap ( which comes from settings.xml )
+     *
+     * @param wagon
+     * @param repositoryId
+     * @param settings
+     * @param container
+     * @param log
+     * @throws TransferFailedException
+     * @todo Remove when {@link WagonManager#getWagon(Repository) is available}. It's available in Maven 2.0.5.
+     */
+    public static void configureWagon(
+        final Wagon wagon,
+        final String repositoryId,
+        final Settings settings,
+        final PlexusContainer container,
+        final Log log )
+    throws TransferFailedException {
+        log.debug( " configureWagon " );
+
+        for ( final Server server : settings.getServers() ) {
+            final String id = server.getId();
+            if ( log.isDebugEnabled() ) {
+                log.debug( "configureWagon server " + id );
+            }
+
+            if ( id != null && id.equals( repositoryId ) && ( server.getConfiguration() != null ) ) {
+                final PlexusConfiguration plexusConf = new XmlPlexusConfiguration( (Xpp3Dom) server.getConfiguration() );
+
+                ComponentConfigurator componentConfigurator = null;
+                try {
+                    componentConfigurator = (ComponentConfigurator) container.lookup( ComponentConfigurator.ROLE, "basic" );
+                    componentConfigurator.configureComponent( wagon, plexusConf, container.getContainerRealm() );
+                } catch ( final ComponentLookupException e ) {
+                    throw new TransferFailedException(
+                        "While configuring wagon for \'" + repositoryId + "\': Unable to lookup wagon configurator."
+                        + " Wagon configuration cannot be applied.", e );
+                } catch ( ComponentConfigurationException e ) {
+                    throw new TransferFailedException( "While configuring wagon for \'" + repositoryId
+                                                       + "\': Unable to apply wagon configuration.", e );
+                } finally {
+                    if ( componentConfigurator != null ) {
+                        try {
+                            container.release( componentConfigurator );
+                        } catch ( ComponentLifecycleException e ) {
+                            log.error( "Problem releasing configurator - ignoring: " + e.getMessage() );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    // Supports only Maven >= 3.0
+    public static ProxyInfo getProxyInfo( final MavenSession mavenSession, final Log log, final Repository repository, final SettingsDecrypter settingsDecrypter ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getProxyInfo(MavenSession, Log, Repository, SettingsDecrypter)' has an NPath complexity of 252](https://app.codacy.com/manual/asciidoctor-lifecycle-maven-plugin/asciidoctor-lifecycle-maven-plugin/pullRequest?prid=5390026)